src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -98,7 +104,26 @@ public int getHttpStatus() {
   }
 
   public boolean isSuccess() {
-    return successCodes.contains(getHttpStatus());
+    boolean isSuccess = false;
+    // ServiceNow Rest API may return 200 OK response with an error message in the response body.
+    // Normally we would expect non-200 response code if there's an error.
+    if (this.isRetryable) {
+      throw new RetryableException();
```

Comment:
Sorry, I just noticed this. We shouldn't be throwing an exception in this method - just return true/false.
return isRetryable || successCodes.contains(getHttpStatus());

In the caller, if isSuccess() returns false, throw RetryableException if the error is retryable (add a getter for isRetryable)

Reply:
Removed the exception from `isSuccess()` method

Added a getter `isRetryable()` in `RestAPIResponse` class. Using this getter to check the value of `isRetryable`  Throwing `RetryableException` if the value is `true`

This is how it looks:
```
public boolean isSuccess() {
   boolean isSuccess = false;
   // ServiceNow Rest API may return 200 OK response with an error message in the response body.
   // Normally we would expect non-200 response code if there's an error.
   if (this.isRetryable) {
          isSuccess = false;
   } else if (successCodes.contains(getHttpStatus())) {
          isSuccess = true;
   }
   return isSuccess;
}
```

If we use this : `return isRetryable || successCodes.contains(getHttpStatus());`, then when `isRetryable`is `true`, the `isSuccess()` would be true and the below `if` won't get executed and thus retry logic won't be invoked.

```
if (!apiResponse.isSuccess()) {
        if (apiResponse.isRetryable()) {
          throw new RetryableException();
        }
        return Collections.emptyList();
      }
```